smithy-model/src/main/java/software/amazon/smithy/model/neighbor/NeighborVisitor.java

```diff
@@ -91,11 +91,22 @@ private void addBound(List<Relationship> result, Shape container, ShapeId bindin
         // Add all operation BINDING relationships (resource -> operation) and BOUND relations (operation -> resource).
         shape.getAllOperations().forEach(id -> addBinding(result, shape, id, RelationshipType.OPERATION));
         // Do the same, but for all of the lifecycle operations. Note: does not yet another BOUND relationships.
-        shape.getCreate().ifPresent(id -> result.add(relationship(shape, RelationshipType.CREATE, id)));
         shape.getRead().ifPresent(id -> result.add(relationship(shape, RelationshipType.READ, id)));
         shape.getUpdate().ifPresent(id -> result.add(relationship(shape, RelationshipType.UPDATE, id)));
         shape.getDelete().ifPresent(id -> result.add(relationship(shape, RelationshipType.DELETE, id)));
-        shape.getList().ifPresent(id -> result.add(relationship(shape, RelationshipType.LIST, id)));
+        shape.getPut().ifPresent(id -> result.add(relationship(shape, RelationshipType.PUT, id)));
+        // LIST and CREATE are, by definition, collection operations
+        shape.getCreate().ifPresent(id -> {
+            result.add(relationship(shape, RelationshipType.CREATE, id));
+            result.add(relationship(shape, RelationshipType.COLLECTION_OPERATION, id));
```

Comment:
This behavior differs from how the OPERATION relationship is added. The OPERATION relationship is only formed for operations literally bound to the "operations" property, whereas the handling here for COLLECTION_OPERATION is to also emit this relationship for CREATE and LIST. We should probably make this work like the OPERATION binding.

Reply:
The OPERATION is bound to all operations, not just those in the `operations` key:

```
// Add all operation BINDING relationships (resource -> operation) and BOUND relations (operation -> resource).
shape.getAllOperations().forEach(id -> addBinding(result, shape, id, RelationshipType.OPERATION));
// Do the same, but for all of the lifecycle operations. Note: does not yet another BOUND relationships.
```

It would need to be `getOperations()` for it to only apply to what's in the `operations` key. Is that a bug? The comment immediately below seems to imply it's not, but then the docs in the relationship type imply it is.

Reply:
Ah, good point. This needs some cleanup, and it looks like you also need to update the selectors docs: https://awslabs.github.io/smithy/spec/selectors.html#relationships

Based on the selector docs, I think it makes sense to have OPERATION continue to mean all operations (including collection operations), and we should add PUT.

This leaves COLLECTION_OPERATION as a bit of an outlier; we don't currently disambiguate between collection and instance operation bindings today since the OPERATION relationship includes both instance and collection operations. If we need to add this kind of relationship in the future, we could potentially add "OPERATIONS" to mean specifically operations bound to the "operations" property, and "COLLECTION_OPERATIONS" to mean operations bound to the "collectionOperations" property.

Thoughts?

Reply:
We discussed this offline, and came to the decision to leave COLLECTION_OPERATION as is and also add in INSTANCE_OPERATION. You might want to, for instance, create traits that can only target one or the other.